### DISCUSSION TITLE
Run an IOC server without the ioc_arg_parser function

### DISCUSSION THREAD

**Discussion by wigging** [caproto/caproto#835] [Q&A] [answered]
I can run the simple IOC example (shown below) without any problems.

```python
from caproto.server import PVGroup, ioc_arg_parser, pvproperty, run

class SimpleIOC(PVGroup):
    A = pvproperty(value=1, doc="An integer")
    B = pvproperty(value=2.0, doc="A float")
    C = pvproperty(value=[1, 2, 3], doc="An array of integers (max length 3)")

if __name__ == "__main__":
    ioc_options, run_options = ioc_arg_parser(default_prefix="simple:", desc="Example")
    ioc = SimpleIOC(**ioc_options)
    run(ioc.pvdb, **run_options)
``` 

This outputs the following in the terminal:

```
[I 21:27:54.150       server:  152] Asyncio server starting up...
[I 21:27:54.150       server:  159] Listening on 0.0.0.0:5064
[I 21:27:54.151       server:  196] Server startup complete.
```

I would like to run this example without using the `ioc_arg_parser` function. I tried the approach shown below which seems to run the server but none of the log info is shown in the terminal.

```python
from caproto.server import PVGroup, pvproperty, run

class SimpleIOC(PVGroup):
    A = pvproperty(value=1, doc="An integer")
    B = pvproperty(value=2.0, doc="A float")
    C = pvproperty(value=[1, 2, 3], doc="An array of integers (max length 3)")

if __name__ == "__main__":
    ioc = SimpleIOC("simple")
    run(ioc.pvdb)
```

How can I run the IOC server without the `ioc_arg_parser` function and still get the log output?

**Comment by tacaswell** [comment, marked as answer]
The logger handlers are set up at https://github.com/caproto/caproto/blob/46020e4816c7dbe8ccae9b7f0af362ce9a7f4f07/caproto/server/server.py#L2182-L2211 and called at https://github.com/caproto/caproto/blob/46020e4816c7dbe8ccae9b7f0af362ce9a7f4f07/caproto/server/server.py#L2250-L2253.

If you want the logging that is the logic / configuration you will need to extract and run some other way.

My counter question is why do you not want to use `ioc_arg_parser` ?  If you want to add extra parameters to the CLI see examples like https://github.com/caproto/caproto/blob/46020e4816c7dbe8ccae9b7f0af362ce9a7f4f07/caproto/ioc_examples/formula_ioc.py#L153-L171 that are a bit more verbose but lets you use the standard flags (like `-vvvv` to see what is going !) without much extra work.

**Reply by wigging** [reply]
I don't want to use the `ioc_arg_parser` function because I'm trying to include an IOC server in a package that already has a CLI via the `argparse` library. Calling `_set_handler_with_logger(logger_name='caproto.ctx', level='INFO')` in my code does the trick. But I guess that logger function is meant to be private based on the underscore in the function name.
